docs/src/modules/utils/generateMarkdown.js

```diff
@@ -273,6 +273,10 @@ function generateProps(reactAPI) {
     return textProps;
   }, text);
 
+  text = `${text}\nThe \`ref\` is ${
+    reactAPI.forwardsRefTo == null ? '**not** ' : ''
+  }forwarded to the root element.\n`;
```

Comment:
We probably need more granular control of this. I'm not sure this is always the "root element" (in any definition proposed in #15138).

Reply:
We have one case where it's not the case <ListItem button />. We might have another one with InputBase.

Reply:
`ListItem` is actually a component where I would consider the current behavior a bug. But it's a component that is good example where "root = outermost" definition breaks.

`InputBase` looks good to me.

We can improve the test for this behavior with one of your suggestions from https://github.com/mui-org/material-ui/issues/15138#issuecomment-478515574

Reply:
> InputBase looks good to me.

I mean, if we move #14580 forward.